src/components/render-page/iframe-markdown-renderer.tsx

```diff
@@ -27,26 +27,50 @@ export const IframeMarkdownRenderer: React.FC = () => {
 
   const communicator = useRendererToEditorCommunicator()
 
-  const countWordsInRenderedDocument = useCallback(() => {
-    const documentContainer = document.querySelector('[data-word-count-target]')
-    communicator.sendMessageToOtherSide({
-      type: CommunicationMessageType.ON_WORD_COUNT_CALCULATED,
-      words: documentContainer ? countWords(documentContainer) : 0
-    })
-  }, [communicator])
+  const sendScrolling = useRef<boolean>(false)
+
+  useRendererReceiveHandler(
+    CommunicationMessageType.DISABLE_RENDERER_SCROLL_SOURCE,
+    useCallback(() => {
+      sendScrolling.current = false
+    }, [])
+  )
+
+  useRendererReceiveHandler(
+    CommunicationMessageType.SET_BASE_CONFIGURATION,
+    useCallback((values) => setBaseConfiguration(values.baseConfiguration), [])
+  )
 
-  useRendererReceiveHandler(CommunicationMessageType.SET_BASE_CONFIGURATION, (values) =>
-    setBaseConfiguration(values.baseConfiguration)
+  useRendererReceiveHandler(
+    CommunicationMessageType.SET_MARKDOWN_CONTENT,
+    useCallback((values) => setMarkdownContentLines(values.content), [])
   )
-  useRendererReceiveHandler(CommunicationMessageType.SET_MARKDOWN_CONTENT, (values) =>
-    setMarkdownContentLines(values.content)
+
+  useRendererReceiveHandler(
+    CommunicationMessageType.SET_DARKMODE,
+    useCallback((values) => setDarkMode(values.activated), [])
   )
-  useRendererReceiveHandler(CommunicationMessageType.SET_DARKMODE, (values) => setDarkMode(values.activated))
-  useRendererReceiveHandler(CommunicationMessageType.SET_SCROLL_STATE, (values) => setScrollState(values.scrollState))
-  useRendererReceiveHandler(CommunicationMessageType.SET_FRONTMATTER_INFO, (values) =>
-    setFrontmatterInfo(values.frontmatterInfo)
+
+  useRendererReceiveHandler(
+    CommunicationMessageType.SET_SCROLL_STATE,
+    useCallback((values) => setScrollState(values.scrollState), [])
+  )
+
+  useRendererReceiveHandler(
+    CommunicationMessageType.SET_FRONTMATTER_INFO,
+    useCallback((values) => setFrontmatterInfo(values.frontmatterInfo), [])
+  )
+
+  useRendererReceiveHandler(
+    CommunicationMessageType.GET_WORD_COUNT,
+    useCallback(() => {
+      const documentContainer = document.querySelector('[data-word-count-target]')
+      communicator.sendMessageToOtherSide({
+        type: CommunicationMessageType.ON_WORD_COUNT_CALCULATED,
+        words: documentContainer ? countWords(documentContainer) : 0
```

Comment:
Is the `countWords` method blocking the receive handler when operating on a huge content? Or should we wrap it inside a promise?

Reply:
It shouldn't make any difference because every message is triggering a new event.

Reply:
I guess? @ErikMichelson ?